### PR TITLE
ci: Automate releasing via bob

### PR DESCRIPTION
**Pull request by radeksimko**
## Context

Releasing currently requires a number of additional steps outside of GitHub. For a simple project with simple release pipeline such as this one, there is no strong benefit to do multi-staged releasing, as some other bigger projects do.

## Proposal

Simplify release process by moving promotion steps into a GHA workflow.

## TODO

 - [x] Create a dedicated GH bot account with appropriate access
 - [x] Grant access for `tf-editor-exp-ci-bot` to `bob` repo
 - [x] Generate new PAT within the GH bot account ^
 - [x] Set PAT as a new secret within this repo `BOB_GITHUB_TOKEN`
 - [x] Update release docs

### CODE REVIEW

**Comment by dbanck** [comment]
Can we run the staging and production back to back? Or do we have to wait until staging completes before running production?

**Comment by radeksimko** [comment]
> Can we run the staging and production back to back? Or do we have to wait until staging completes before running production?

I have added the dependency on line 49 https://github.com/hashicorp/terraform-ls/pull/1167/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R49 but it's not strictly necessary. The real "awkward part" though about the whole process is that both `bob` commands are only triggers and do not wait for completion, so the dependency won't make that much of a difference in reality.

I discussed this problem more with the RelEng team and they are planning to remove the requirement of `staging` altogether some time in 2024, which should eventually resolve this awkwardness.

Strictly speaking - and to answer your question - it should be fine because the `production` workflow does not require `staging` to finish until it is actually approved.

**Comment by dbanck** [comment]
Thanks for the additional context. Should we add a note to the README that we need to wait until the staging workflow has succeeded and the artifacts have landed in the staging bucket (also a good time to manually test the build) before approving the production workflow?

**Comment by radeksimko** [comment]
@dbanck added a step for this in the instructions, PTAL

File: .github/CONTRIBUTING.md
Comment (dbanck) [inline .github/CONTRIBUTING.md:211]: I think this isn't required anymore. You only have to approve the production workflow once.

**Review by dbanck — APPROVED** [review]
Nice!

**Pull request merged**
